lib/faktory.rb

```diff
@@ -8,6 +8,8 @@
 require 'faktory/job'
 require 'faktory/connection'
 
+require 'active_job/queue_adapters/faktory_adapter'
```

Comment:
This is something Rails should do automatically for us, right?  We definitely don't want to eager load it.

Reply:
That is to say, FWR should be usable without Rails.

Reply:
Without this line my test rails app is unable to find the adapter and this happens when I try to queue a job.

```
NameError (uninitialized constant ActiveJob::QueueAdapters::FaktoryAdapter)
```

Rails auto-loads the built in adapters like this: https://github.com/rails/rails/blob/master/activejob/lib/active_job/queue_adapters.rb#L112-L122

I'm not sure how to accomplish something similar without causing AJ itself to load. Pointers appreciated.

Reply:
Just lurking here, but in the past, I've used this approach in situations like this:

```ruby
require 'active_job/queue_adapters/faktory_adapter' if defined?(Rails)

# or

require 'active_job/queue_adapters/faktory_adapter' if defined?(ActiveJob)
```

Another option, albeit with more "overhead", would be to move ActiveJob-related code into a standalone gem (i.e. `factory_worker_rails`, `factory_worker_active_job` or similar).

Reply:
`if defined?(Rails)` works perfectly. I've updated the PR to use that. I'm also happy to go a different route if that would be better.

Reply:
I would recommend defined?(::ActiveJob).  People, at times, define ::Rails for various questionable reasons and Rails < 4.2 is also possible.

Reply:
Ok, after trying `defined?(::ActiveJob)` I realized that I didn't test thoroughly enough with `defined?(Rails)`. In both cases I can queue an `ActiveJob` just fine, but in both cases when I start `faktory-worker` it complains about not being able to find `ActiveJob::QueueAdapters::FaktoryAdapter`.

Here's the complete output from `faktory-worker`:

```
2018-03-20T19:50:20.855Z 90748 TID-ovqe2oyg4 ActiveJob::QueueAdapters::FaktoryAdapter::JobWrapper JID-3929f44998f1c708490813b3 INFO: start
2018-03-20T19:50:20.865Z 90748 TID-ovqe2oyg4 ActiveJob::QueueAdapters::FaktoryAdapter::JobWrapper JID-3929f44998f1c708490813b3 INFO: fail: 0.01 sec
2018-03-20T19:50:20.865Z 90748 TID-ovqe2oyg4 WARN: {"context":"Job raised exception","job":{"jid":"3929f44998f1c708490813b3","queue":"default","jobtype":"ActiveJob::QueueAdapters::FaktoryAdapter::JobWrapper","args":[{"arguments":[234],"executions":0,"job_class":"VerboseActiveJob","job_id":"d810f0e4-e674-417e-bb2e-52552414cfb8","locale":"en","priority":null,"provider_job_id":"3929f44998f1c708490813b3","queue_name":"default"}],"created_at":"2018-03-20T19:50:20.854406Z","enqueued_at":"2018-03-20T19:50:20.854412Z","custom":{"wrapped":"VerboseActiveJob"}}}
2018-03-20T19:50:20.865Z 90748 TID-ovqe2oyg4 WARN: NameError: uninitialized constant ActiveJob::QueueAdapters::FaktoryAdapter
2018-03-20T19:50:20.865Z 90748 TID-ovqe2oyg4 WARN: /Users/jgreen/projects/faktory_worker_ruby/lib/faktory/processor.rb:171:in `block in constantize'
/Users/jgreen/projects/faktory_worker_ruby/lib/faktory/processor.rb:170:in `each'
/Users/jgreen/projects/faktory_worker_ruby/lib/faktory/processor.rb:170:in `inject'
/Users/jgreen/projects/faktory_worker_ruby/lib/faktory/processor.rb:170:in `constantize'
/Users/jgreen/projects/faktory_worker_ruby/lib/faktory/processor.rb:133:in `block (3 levels) in dispatch'
/Users/jgreen/projects/faktory_worker_ruby/lib/faktory/rails.rb:24:in `block in call'
/Users/jgreen/.rvm/gems/ruby-2.5.0/gems/activesupport-5.1.5/lib/active_support/execution_wrapper.rb:85:in `wrap'
/Users/jgreen/.rvm/gems/ruby-2.5.0/gems/activesupport-5.1.5/lib/active_support/reloader.rb:68:in `block in wrap'
/Users/jgreen/.rvm/gems/ruby-2.5.0/gems/activesupport-5.1.5/lib/active_support/execution_wrapper.rb:85:in `wrap'
/Users/jgreen/.rvm/gems/ruby-2.5.0/gems/activesupport-5.1.5/lib/active_support/reloader.rb:67:in `wrap'
/Users/jgreen/projects/faktory_worker_ruby/lib/faktory/rails.rb:23:in `call'
/Users/jgreen/projects/faktory_worker_ruby/lib/faktory/processor.rb:132:in `block (2 levels) in dispatch'
/Users/jgreen/projects/faktory_worker_ruby/lib/faktory/job_logger.rb:7:in `call'
/Users/jgreen/projects/faktory_worker_ruby/lib/faktory/processor.rb:127:in `block in dispatch'
/Users/jgreen/projects/faktory_worker_ruby/lib/faktory/logging.rb:43:in `with_context'
/Users/jgreen/projects/faktory_worker_ruby/lib/faktory/logging.rb:37:in `with_job_hash_context'
/Users/jgreen/projects/faktory_worker_ruby/lib/faktory/processor.rb:126:in `dispatch'
/Users/jgreen/projects/faktory_worker_ruby/lib/faktory/processor.rb:145:in `process'
/Users/jgreen/projects/faktory_worker_ruby/lib/faktory/processor.rb:91:in `process_one'
/Users/jgreen/projects/faktory_worker_ruby/lib/faktory/processor.rb:74:in `run'
/Users/jgreen/projects/faktory_worker_ruby/lib/faktory/util.rb:16:in `watchdog'
/Users/jgreen/projects/faktory_worker_ruby/lib/faktory/util.rb:25:in `block in safe_thread'
```

Reply:
Just doing a vanilla require (with no conditional) gets `faktory-worker` running again...